src/bench/streams_findbyte.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <streams.h>
+#include <test/util/setup_common.h>
+
+static void FindByte(benchmark::Bench& bench)
+{
+    // Setup
+    FILE* file = fsbridge::fopen("streams_tmp", "w+b");
```

Comment:
Could we maybe use something like `memfd_create(2)` for the benchmark, to decrease I/O noise?
Anyone knows if there is a windows equivalent or a higher abstraction in boost::filesystem?

Reply:
`fmemopen(3)` also looks interesting

Reply:
Maybe this can give us that? https://www.boost.org/doc/libs/1_72_0/libs/iostreams/doc/classes/mapped_file.html

Reply:
Thanks for the suggestions, but I don't think this matters, because the file IO (read) occurs only on the very first benchmark loop iteration; after that, the data is in memory and there is no IO at all. Each iteration repositions the stream pointer to zero (`bf.SetPos(0)`) and then searches forward for the value 1 (`bf.FindByte(1)`), which is 200 bytes away. But after the first iteration, all of these bytes are in memory, and we're just moving the position between 0 and 200. 

The reason for the 200, by the way, is that with random data, searching for a random byte value the average distance would be half of 256. But the data in `blk.dat` files (which is the only use of this code currently) isn't quite random; zero and `0xff` occur more often (and we never search for those values). So I guessed that 200 is close to a typical distance that this function would move through before finding the requested byte. Maybe I should explain these points in a comment in `bench/streams_findbyte.cpp`.